typegate/tests/e2e/upgrade/upgrade_test.ts

```diff
@@ -0,0 +1,264 @@
+// Copyright Metatype OÜ, licensed under the Elastic License 2.0.
+// SPDX-License-Identifier: Elastic-2.0
+
+import { Meta } from "test-utils/mod.ts";
+import { PUBLISHED_VERSION } from "@dev/consts.ts";
+import { projectDir } from "@dev/utils.ts";
+import { $ } from "@dev/deps.ts";
+import { download } from "download";
+import { Untar } from "std/archive/untar.ts";
+import { readerFromIterable } from "std/streams/mod.ts";
+import { copy } from "std/io/copy.ts";
+import { encodeBase64 } from "std/encoding/base64.ts";
+import { ProcessOutputLines } from "test-utils/process.ts";
+import { newTempDir } from "test-utils/dir.ts";
+import { transformSyncConfig } from "@typegate/config.ts";
+import { clearSyncData, setupSync } from "test-utils/hooks.ts";
+import { assertEquals } from "std/assert/assert_equals.ts";
+
+const previousVersion = PUBLISHED_VERSION;
+
+const tempDir = $.path(projectDir).join("tmp");
+
+function getAssetName(version: string) {
+  return `meta-cli-v${version}-${Deno.build.target}`;
+}
+
+const syncEnvs = {
+  SYNC_REDIS_URL: "redis://:password@localhost:6379/12",
+  SYNC_S3_HOST: "http://localhost:9000",
+  SYNC_S3_REGION: "local",
+  SYNC_S3_ACCESS_KEY: "minio",
+  SYNC_S3_SECRET_KEY: "password",
+  SYNC_S3_BUCKET: "upgrade-test",
+};
+
+const syncConfig = transformSyncConfig({
+  redis_url: new URL(syncEnvs.SYNC_REDIS_URL),
+  s3_host: new URL(syncEnvs.SYNC_S3_HOST),
+  s3_region: syncEnvs.SYNC_S3_REGION,
+  s3_access_key: syncEnvs.SYNC_S3_ACCESS_KEY,
+  s3_secret_key: syncEnvs.SYNC_S3_SECRET_KEY,
+  s3_bucket: syncEnvs.SYNC_S3_BUCKET,
+  s3_path_style: true,
+});
+
+// TODO remove after the next release
+// These typegates are disabled because a compatibity issue on the pyrt wasm:
+//  Module was  WebAssembly backtrace support but it is enabled for the host
+const disabled = [
+  "quick-start-project.ts",
+  "faas-runner.ts",
+  "microservice-orchestration.ts",
+  "metagen-rs.ts",
+  "metagen-py.ts",
+];
+
+async function checkMetaBin(path: typeof tempDir) {
+  try {
+    if (!(await path.exists())) {
+      return false;
+    }
+    const res = await $`bash -c 'meta --version'`
+      .env("PATH", `${path.parent()!.toString()}:${Deno.env.get("PATH")}`)
+      .stdout("piped");
+    console.log(res.stdout);
+    return true;
+  } catch (_e) {
+    return false;
+  }
+}
+
+// download the fat version of the cli on the latest stable release
+async function downloadAndExtractAsset(version: string) {
+  const name = getAssetName(version);
+  const extractTargetDir = tempDir.join(name);
+  const metaBin = extractTargetDir.join("meta");
+  if (await checkMetaBin(metaBin)) {
+    return metaBin.toString();
```

Comment:
The path here might resolve to a global `meta` installation. Try executing the file at the path instead of manipulating path vars? You can use $.raw`` if you're having issues with dax.

Reply:
I have Deno permission issues with that...

Reply:
Hmm. I wonder if `allow-run` takes globs.